main.go

```diff
@@ -1,13 +1,25 @@
 package main
 
 import (
+	"flag"
 	"github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
 )
 
 //go:generate go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
 
 func main() {
-	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: octopusdeploy.Provider})
+	var debugMode bool
+	flag.BoolVar(&debugMode, "debug", false, "set to true to run the provider with support for debuggers like delve")
+	flag.Parse()
+
+	opts := &plugin.ServeOpts{
+		ProviderFunc: octopusdeploy.Provider}
+
+	if debugMode {
+		opts.Debug = true
+		opts.ProviderAddr = "octopus.com/com/octopusdeploy"
```

Comment:
My guess is that this _might_ fail if the provider is pulled from the Terraform Registry -- it uses a different address.

Reply:
I follow the instructions in https://opencredo.com/blogs/running-a-terraform-provider-with-a-debugger/.
And it works 😄 

So @jbristowe, this is only the address to be used when we start with `--debug` argument, which would be done using the local copy.

I hope this makes sense, let me know?

Reply:
In other words, it wouldn't make sense to try to attach the debugger to a non-local compiled version.